runtime/src/kmp.h

```diff
@@ -3682,7 +3686,11 @@ KMP_EXPORT void __kmpc_taskloop(ident_t *loc, kmp_int32 gtid, kmp_task_t *task,
                                 kmp_int32 if_val, kmp_uint64 *lb,
                                 kmp_uint64 *ub, kmp_int64 st, kmp_int32 nogroup,
                                 kmp_int32 sched, kmp_uint64 grainsize,
-                                void *task_dup);
+                                void *task_dup
+#if OMPT_SUPPORT && OMPT_OPTIONAL
+                                 , void * codeptr
```

Comment:
Is this argument needed and used?

Reply:
In reviewing not just this line but the whole file, all of the locking routines now have 

#if OMPT_SUPPORT && OMPT_OPTIONAL
                               , void * codeptr
#endif

That shouldn't be the case if these routines are called directly by compiler-generated code. We aren't planning to change the API used by the compilers.

If these routines are called by the compilers, there is no point in passing the return address as the codeptr argument when the return address is readily available in the caller (e.g. it is as the top of stack on x86 and in link registers on PPC and ARM).